### PR TITLE
chore: Merge Slurm-infrastructure support changes job.go

### PR DESCRIPTION
**Pull request by jerryharrow**
Merge in new access methods needed by Slurm RM.

## Description

Merge in new access methods needed by Slurm RM.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by jerryharrow** [comment]
Ready for review @stoksc

File: master/internal/resourcemanagers/job.go [hunk 1]
```diff
@@ -51,7 +50,26 @@ func jobStats(taskList *taskList) *jobv1.QueueStats {
 		}
 		reqs = append(reqs, req)
 	}
+	stats := requestsToQueueStats(reqs)
+	return stats
+}
+
+//nolint:deadcode // Method used by Slurm support in determined-ee
+func jobStatsByPool(taskList *taskList, resourcePool string) *jobv1.QueueStats {
```
Comment (stoksc) [inline master/internal/resourcemanagers/job.go:58]: maybe better (more flexible) is just `jobStatsByPool` + `taskList.byPool(name string)` but this is definitely fine

Comment (jerryharrow) [reply master/internal/resourcemanagers/job.go:58]: Don't understand your comment.   You mean more like this?

//nolint:deadcode // Method jobStatsByPool is used by Slurm support in determined-ee

Comment (stoksc) [reply master/internal/resourcemanagers/job.go:58]: oh I just meant, we wouldn't need `jobStatsByPool` if `taskList.byPool(name string)` existed. sorry, total non-comment. was just thinking `taskList` could probably use the enhancement

**Pull request merged**